src/amo/pages/UserProfileEdit/index.js

```diff
@@ -855,12 +855,15 @@ export class UserProfileEditBase extends React.Component<Props, State> {
 
 export function mapStateToProps(state: AppState, ownProps: Props) {
   const { clientApp, lang } = state.api;
+
   const { params } = ownProps.match;
+  const userId = Number(params.userId);
 
   const currentUser = getCurrentUser(state.users);
-  const user = params.username
-    ? getUserByUsername(state.users, params.username)
-    : currentUser;
+  const user =
+    params.userId && !Number.isNaN(userId)
```

Comment:
So if `Number(params.userId)` is `NaN`, we're going to default `user` to `currentUser`. Is that really what we want? I don't think that replicates the previous logic.

Reply:
ah, right